pkg/asyncdr/asyncdr.go

```diff
@@ -756,3 +761,110 @@ func CheckDefaultPxStorageClass(name string) error {
 	}
 	return nil
 }
+
+func ChangePxServiceToLoadBalancer(internalLB bool) error {
+	// Check if service is already of type loadbalancer
+	const (
+		pxStcServiceTypeKey  = "portworx.io/service-type"
+		stcLoadBalancerValue = "portworx-service:LoadBalancer"
+		pxStcServiceKey      = "service/portworx-service"
+		awsInternalLBKey     = "service.beta.kubernetes.io/aws-load-balancer-internal"
+		awsInternalLBValue   = "true"
+		awsLBTypeKey         = "service.beta.kubernetes.io/aws-load-balancer-type"
+		awsLBTypeVal         = "nlb"
+		awsNLBTargetTypeKey  = "service.beta.kubernetes.io/aws-load-balancer-nlb-target-type"
+		awsNLBTargetTypeVal  = "ip"
+		awsLBSubnetKey       = "service.beta.kubernetes.io/aws-load-balancer-subnets"
+		pxServiceName        = "portworx-service"
+		pxNamespace          = "kube-system"
+	)
+
+	pxService, err := core.Instance().GetService(pxServiceName, pxNamespace)
+	if err != nil {
+		return fmt.Errorf("failed to get portworx service before changing it to load balancer: %v", err)
+	}
+	log.Infof("Current Service Type is %v, pxService.Spec.Type")
+	if pxService.Spec.Type == v1.ServiceTypeLoadBalancer {
+		log.InfoD("portworx service is already of type LoadBalancer")
+		return nil
+	}
+
+	storageDriverName := flag.Lookup("storage-driver").Value.(flag.Getter).Get().(string)
+	log.Infof("Storage driver name: %v", storageDriverName)
+
+	if storageDriverName == storkdriver.PortworxDriverName {
+		stc, err := operator.Instance().ListStorageClusters(pxNamespace)
```

Comment:
Inst().V.GetDriver() will give you the STC

Reply:
Done.

Reply:
you are still using  operator.Instance()

Reply:
Sorry I missed this, just updated please check

Reply:
Done.